app/models/service_template_transformation_plan/validate_config_info.rb

```diff
@@ -14,18 +14,32 @@ def validate_config_info(options)
                 else
                   config_info[:transformation_mapping]
                 end
-
+      # mapping.invalid? added as a part of validating migration plan
       raise _('Must provide an existing transformation mapping') if mapping.blank?
+      raise _('Transformation mapping was invalid.  Provide a valid one') if mapping.invalid?
 
       pre_service_id  = config_info[:pre_service].try(:id) || config_info[:pre_service_id]
       post_service_id = config_info[:post_service].try(:id) || config_info[:post_service_id]
 
+      # Add as a part of validating migration plan
+      # check service_type. Filter type=ServiceTemplateAnsiblePlaybook
+      if pre_service_id.present?
+        pre_service_type = ServiceTemplate.find(pre_service_id).type
+        raise _('Premigration service type MUST be \"ServiceTemplateAnsiblePlaybook\"') if pre_service_type != "ServiceTemplateAnsiblePlaybook"
```

Comment:
We should also check that `ServiceTemplate.find(pre_service_id)` isn't nil. Similar to what it done at L17.
Same for `post_service_id`.

Reply:
`ServiceTemplate.find(pre_service_id)` would already raise an error if not found.  It can't return nil .

Reply:
If `pre_service_id` is not `nil`, it doesn't mean that `ServiceTemplate.find(pre_service_id)` will not return `nil`. And in that case, it will throw `undefined method 'type' for nil:NilClass`.

Reply:
True, agreed, and implemented.